### PR TITLE
2789 integrate telemetry module

### LINKED ISSUE #2789: Integrate telemetry module
Integrate API errors, records creation count and dataset publish actions with the telemetry module

### PR DESCRIPTION
**Pull request by frascuchon**
# Description

This PR aligns the current telemetry with new feedback datasets by tracking:

- Dataset publish action
- #records when new records are created
- Server error when something is wrong

### CODE REVIEW

**Review by dvsrepo — APPROVED** [review]
LGTM!

File: src/argilla/server/commons/telemetry.py [hunk 1]
```diff
@@ -95,31 +95,31 @@ def _process_request_info(request: Request):
...
         {
             "is_default_user": username == "argilla",
             "user_hash": str(uuid.uuid5(namespace=telemetry_client.server_id, name=username)),
             **_process_request_info(request),
         },
     )
+
+
+def track_data(event: str, data: dict):
+    global telemetry_client
```
Comment (gabrielmbmb) [inline src/argilla/server/commons/telemetry.py]: We can remove this line if we're not going to set a value in this function for the variable `telemetry_client`

**Comment by gabrielmbmb** [comment]
LGTM! Just some minor comment

**Comment by frascuchon** [comment]
@gabrielmbmb I did some code refactoring regarding the `track_data` function and the `global` reference

**Review by gabrielmbmb — APPROVED** [review]
LGTM!

**Pull request merged**